### ISSUE TITLE
cover art in info area sometimes doesn't show up

### ISSUE THREAD

**Issue by strahlc** [exaile/exaile#215]
In some songs, the embedded album art is not displayed in the InfoArea. The Desktop Cover plugin also doesn't show the image of the affected tracks.
The properies dialog show the cover art correctly.
On the console I see no error messages.
Affects current master ( 8655a17)

**Comment by strahlc** [comment]
affects current master ( 8655a17)

**Comment by strahlc** [comment]
Sorry, I can reproduce this bug only with one file so far.
Other mp3 files with embedded PNG cover art doesn't have this problem.

**Comment by virtuald** [comment]
@strahlc do you happen to remember if these were ogg files? I just found a bug related to that I think...

**Comment by virtuald** [comment]
https://github.com/exaile/exaile/issues/331 may be related

**Comment by strahlc** [comment]
@virtuald no, this happens with mp3 files.

**Comment by sbrubes** [comment]
It could happen and from my tests it's related to the associated _db_string_ for _track_.

I'll not add steps to reproduce because the solution it's easier than explaning. But if any of you insist, I could do.

https://github.com/exaile/exaile/blob/master/xl/covers.py#L307
 ```
        if db_string:
            cover = self.get_cover_data(db_string, use_default=False)
            if cover: return cover
```
If could not get it, continues and tries to find it. https://github.com/exaile/exaile/blob/master/xl/covers.py#L313

**Comment by virtuald** [comment]
@sbrubes that makes sense... though, it looks like if a cover is found on lines 318, then `set_cover` will store the cover in the db, which would cause 306/308 to return something the next time it was called?

It does seem a bit convoluted, so it could use some refactoring.

**Comment by sbrubes** [comment]
Yes, it is like:

> 
> found in cache: return it,
> not found, search and add.
> 

But from my use, "'cover' as tag" it just index a path to file (where tag is), to catch it.

**Comment by virtuald** [comment]
Fixed in #475
